src/graphapi/conv_bias_res_add_activ_forward_executor.cpp

```diff
@@ -0,0 +1,131 @@
+#include <miopen/graphapi/conv_bias_res_add_activ_forward_executor.hpp>
+#include <miopen/fusion.hpp>
+#include <miopen/visit_float.hpp>
+
+namespace miopen {
+
+namespace graphapi {
+
+static std::vector<int> Convert(const std::vector<int64_t>& values)
+{
+    return {values.begin(), values.end()};
+}
+
+static ConvolutionDescriptor Convert(const Convolution& conv, int groupCount)
+{
+    return {conv.getSpatialDims(),
+            conv.getMode(),
+            miopenPaddingMode_t::miopenPaddingDefault,
+            Convert(conv.getPrePaddings()),
+            Convert(conv.getFilterStrides()),
+            Convert(conv.getDilations()),
+            Convert(conv.getPostPaddings()),
+            groupCount};
+}
+
+static float Convert(half_float::half value) { return static_cast<float>(value); }
+
+static float Convert(float value) { return value; }
+
+void ConvBiasResAddActivForwardExecutor::execute(miopenHandle_t handle, const VariantPack& vpk)
+{
+    MIOPEN_THROW_IF(mActivationOp->getPointwise()->getMode() !=
+                        miopenPointwiseMode_t::MIOPEN_POINTWISE_RELU_FWD,
+                    "invalid activation operation");
+
+    MIOPEN_THROW_IF(mAddOp->getPointwise()->getMode() !=
+                        miopenPointwiseMode_t::MIOPEN_POINTWISE_ADD,
+                    "invalid pointwise operation for add op");
+
+    MIOPEN_THROW_IF(mBiasOp->getPointwise()->getMode() !=
+                        miopenPointwiseMode_t::MIOPEN_POINTWISE_ADD,
+                    "invalid pointwise operation for bias op");
+
+    auto* xDesc = mConvOp->getX();
+    auto* xData = vpk.getDataPointer(xDesc->getId());
+
+    auto* wDesc = mConvOp->getW();
+    auto* wData = vpk.getDataPointer(wDesc->getId());
+
+    std::size_t in_c  = xDesc->GetLengths()[1];
+    std::size_t wei_c = wDesc->GetLengths()[1];
+    int groupCount    = in_c / wei_c;
+    auto convDesc     = Convert(*mConvOp->getConvolution(), groupCount);
+
+    bool xIsVirtual = mAddOp->getX()->isVirtual();
+    auto* zDesc     = xIsVirtual ? mAddOp->getB() : mAddOp->getX();
+    auto* zData     = vpk.getDataPointer(zDesc->getId());
+
+    auto biasDesc  = mBiasOp->getX()->isVirtual() ? mBiasOp->getB() : mBiasOp->getX();
+    auto* biasData = vpk.getDataPointer(biasDesc->getId());
+
+    auto yDesc  = mActivationOp->getY();
+    auto* yData = vpk.getDataPointer(yDesc->getId());
+
+    auto convertToFloat = [](auto&& arg) { return Convert(arg); };
+    // The virtual tensor for add is the result of the convolution, and combining the alpha1's allow
+    // users to specify it through the graphAPI properly.
+    float alpha1 =
+        mConvOp->getAlpha() * (xIsVirtual ? std::visit(convertToFloat, mAddOp->getAlpha1())
+                                          : std::visit(convertToFloat, mAddOp->getAlpha2()));
+
+    // The non-virtual tensor for add should be alpha 2.
+    float alpha2 =
+        xIsVirtual ? std::get<float>(mAddOp->getAlpha2()) : std::get<float>(mAddOp->getAlpha1());
```

Comment:
Looking at this again, I think it would make more sense to move some of this into the GraphPatternMatcher, and instead of passing the operations, I can pass the values I need out of them.  

For example, instead of passing the addOp, and needing to find which tensor to use for z, I could do that in the GraphPatternMatcher, and then make the executor's constructor explicit.